packages/react-animation/src/carousel/hooks/useCarouselBounds.ts

```diff
@@ -0,0 +1,45 @@
+import { useResizeObserver, useMutationObserver } from '@mediamonks/react-hooks';
+import { type RefObject, useCallback, useState } from 'react';
+import { CarouselType } from './useCarousel.js';
+import { useCarouselCalculations } from './useCarouselCalculations.js';
+
+type CarouselBoundsOptions = {
+  type?: CarouselType;
+};
+export function useCarouselBounds(
+  sliderRef: RefObject<HTMLElement>,
+  { type = CarouselType.X }: CarouselBoundsOptions = {},
+): Draggable.BoundsMinMax | null {
+  const [bounds, setBounds] = useState<Draggable.BoundsMinMax | null>(null);
+  const { getElementOffset } = useCarouselCalculations(type);
+
+  const onUpdateBounds = useCallback(() => {
+    if (sliderRef.current === null) {
+      // eslint-disable-next-line no-console
+      console.warn("Can't set bounds, sliderRef is undefined");
+      return;
+    }
+
```

Comment:
In foundation I added an extra check for the example AutoHeight to work without an insane amount of re-renders. Please see full code changes in foundation.

`if (type === CarouselType.X && containerWidth.current === sliderRef.current.offsetWidth) {
      return;
    }`